### PR TITLE
Updates postgres versions for test images

### PR DESCRIPTION
**Pull request by gurkanindibay**
DESCRIPTION: Updates pg version for test images into 12.9 13.5 and 14.1
* [ ] Change suffix into sha when https://github.com/citusdata/the-process/pull/67 is merged
* [ ] Remove dev tag suffix after approval

### CODE REVIEW

File: .circleci/config.yml
Comment (SaitTalhaNisanci) [inline .circleci/config.yml:662]: I am wondering if we should add these versions to `parameters` section so that we have a single place to update them, for example we could have 3 different parameters such as:

```
pg_12:
   type: string
   default: 12.9
pg_13:
   type: string
   default: 13.5
pg_14:
   type: string
   default: 14.1
pg_upgrade:
   type: string
   default: 12.9-13.5-14.1
```

Then when updating new images, we only need to update them in the parameters section

**Review by SaitTalhaNisanci — COMMENTED** [review]
The changes look sane

File: .circleci/config.yml
Comment (thanodnl) [reply .circleci/config.yml:662]: Would love this, but lets work on this separately from upgrading image versions and not block this PR on such a change.

File: .circleci/config.yml
Comment (thanodnl) [inline .circleci/config.yml:9]: Don't forget to change this after we have landed the CI changes and pushed released images.

**Review by thanodnl — APPROVED** [review]
Looks good, please wait. Although approved, lets land the changes for `the-process` first and update the `image_suffix` before landing this change.

File: .circleci/config.yml
Comment (gurkanindibay) [reply .circleci/config.yml:662]: I tried it but it seems that we can not use parameters under workflow context. And I tried to parameterize image_tag but I couldn't find a method to parameterize it

File: src/test/regress/expected/failure_connection_establishment_0.out
Comment (onderkalaci) [inline src/test/regress/expected/failure_connection_establishment_0.out:1]: maintaining soo many alternative files is very hard. Do we really need all of them?

Comment (gurkanindibay) [reply src/test/regress/expected/failure_connection_establishment_0.out:1]: I see your point but with libpq changes, wach version has its own messages. Therefore yes it is required. I encountered all the combinations for messages. But If you think that message is not correct I can check it

Comment (onurctirtir) [reply src/test/regress/expected/failure_connection_establishment_0.out:1]: As we discussed offline, I guess Onder meant exploring whether making use of `normalize.sed` instead of those alternative outputs could work or not.

**Comment by hanefi** [comment]
I suggest you rebase your changes on top of #5561 that parameterized PG versions in CircleCI config. This will make your changes easier to read.

File: src/test/regress/expected/failure_connection_establishment_0.out
Comment (gurkanindibay) [reply src/test/regress/expected/failure_connection_establishment_0.out]: I tried to use normalize.sed as much as possible. I could eliminate some outputs. I tried creating regex for the changes for the remaining ones however I can not create a general rule without affecting other files.

Comment (gurkanindibay) [reply src/test/regress/expected/failure_connection_establishment_0.out]: If we want to reduce test outputs, I think we can add a regex structure which will be able to work just for one file. By this way, adding a general regex will not affect other files

Comment (hanefi) [reply src/test/regress/expected/failure_connection_establishment_0.out]: IMO, It's ok to change other files. It would be easier to maintain if all matching messages are replaced in all the files.

I think it is safer, and better to update more tests to achieve consistency

Comment (gurkanindibay) [reply src/test/regress/expected/failure_connection_establishment_0.out]: I needed to remove the rule to replace 
``
WARNING:  connection not open
``
with 
```
WARNING:  server closed the connection unexpectedly
	This probably means the server terminated abnormally
	before or while processing the request.
connection not open
```

**Pull request closed without merging**
